core/src/main/java/io/grpc/internal/ClientCallImpl.java

```diff
@@ -304,15 +308,18 @@ public void sendMessage(ReqT message) {
     Preconditions.checkState(stream != null, "Not started");
     Preconditions.checkState(!cancelCalled, "call was cancelled");
     Preconditions.checkState(!halfCloseCalled, "call was half-closed");
-    boolean failed = true;
+    Status failure = Status.CANCELLED;
     try {
       InputStream messageIs = method.streamRequest(message);
       stream.writeMessage(messageIs);
-      failed = false;
+      failure = null;
+    } catch (RuntimeException e) {
```

Comment:
We were re-throwing before to get the exception logged since we weren't passing it in cancel(). It would probably be most appropriate to not re-throw.